saleor/graphql/discount/mutations/sale_base_discount_catalogue.py

```diff
@@ -0,0 +1,100 @@
+from collections import defaultdict
+from typing import DefaultDict, Set
+
+import graphene
+from django.core.exceptions import ValidationError
+
+from ....discount.error_codes import DiscountErrorCode
+from ....discount.utils import CatalogueInfo
+from ....product.tasks import update_products_discounted_prices_of_catalogues_task
+from ....product.utils import get_products_ids_without_variants
+from ...core.mutations import BaseMutation
+from ...product.types import Category, Collection, Product, ProductVariant
+
+NodeCatalogueInfo = DefaultDict[str, Set[str]]
```

Comment:
Instead of importing this type inside all different files, maybe we could just use there `DefaultDict[str, Set[str]]`?